sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -331,6 +332,268 @@ case class RoundCeil(child: Expression, scale: Expression)
+        if (jm.BigInteger.ZERO.compareTo(wholePart) == 0) {
+          new jm.BigDecimal(wholePart)
+        } else {
+          val pow = jm.BigInteger.valueOf(Math.pow(10, Math.abs(position)).toLong)
+          new jm.BigDecimal(wholePart.divide(pow).multiply(pow), 0)
+        }
+      }
+    }
+  }
+}
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """_FUNC_(number[, position]) - Returns the number after truncated to the specified places.
+    An optional `position` parameter can be specified to truncate digits to the right of the decimal point.
+    If 0, it removes all the decimal values and returns only the integer.
+    If negative, the number is truncated to the left side of the decimal point.
+    There is an overloaded version of this function to truncate date values
+    _FUNC_(date, fmt) - Returns `date` with the time portion of the day truncated to the unit specified by the format model `fmt`.
+  """,
+  arguments = """
+    Arguments:
+      * number - number to be truncated
+      * position - number of decimal places up to which the given number is to be truncated
+    Arguments: To truncate date value:
+      * date - date value or valid date string
+      * fmt - the format representing the unit to be truncated to
+          - "YEAR", "YYYY", "YY" - truncate to the first date of the year that the `date` falls in
+          - "QUARTER" - truncate to the first date of the quarter that the `date` falls in
+          - "MONTH", "MM", "MON" - truncate to the first date of the month that the `date` falls in
+          - "WEEK" - truncate to the Monday of the week that the `date` falls in
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('2019-08-04', 'week');
+       2019-07-29
+      > SELECT _FUNC_('2019-08-04', 'quarter');
+       2019-07-01
+      > SELECT _FUNC_('2009-02-12', 'MM');
+       2009-02-01
+      > SELECT _FUNC_('2015-10-27', 'YEAR');
+       2015-01-01
+      > SELECT _FUNC_(-10.11, 0);
+       -10
+      > SELECT _FUNC_(10.11, -1);
+       10
+      > SELECT _FUNC_(100.61, 0);
+       100
+      > SELECT _FUNC_(-19087.1560, -3);
+       -19000
+      > SELECT _FUNC_(10876.5489, -1);
+       10870
+      > SELECT _FUNC_(-7767.1160, 2);
+       -7767.11
+      > SELECT _FUNC_(17646.6019, 3);
+       17646.601
+  """,
+  since = "3.4.0",
+  group = "math_funcs")
+// scalastyle:on line.size.limit
+object TruncExpressionBuilder extends ExpressionBuilder {
+  override def build(funcName: String, expressions: Seq[Expression]): Expression = {
+    val numArgs = expressions.length
+    if (numArgs >= 1) {
+      expressions(0).dataType match {
+        case ByteType |  ShortType | IntegerType | LongType | FloatType | DoubleType
+             | DecimalType.Fixed(_, _) => buildTruncNumber(funcName, expressions)
+        case _ => buildTruncDate(funcName, expressions)
+      }
+    } else {
+      throw QueryCompilationErrors.invalidFunctionArgumentNumberError(Seq(2), funcName, numArgs)
```

Comment:
you can check `if (numArgs < 1)` and throw the error first, de-denting the rest of the block to simplify the code.

Reply:
Done